File: src/main/java/com/server/ordering/domain/Basket.java
package com.server.ordering.domain;

import com.server.ordering.domain.member.Customer;
import lombok.*;
import javax.persistence.*;

import static javax.persistence.FetchType.LAZY;
import static javax.persistence.GenerationType.*;
import static lombok.AccessLevel.*;

@Entity
@Getter
@RequiredArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class Basket {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "basket_id")
    private Long id;

    @NonNull
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @NonNull
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "food_id")
    private Food food;

    @NonNull
    private int price;
    @NonNull
    private int count;

    public static Basket CreateBasket(Customer customer, Food food, int price, int count) {
        return new Basket(customer, food, price, count);
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/Basket.kt
package com.byeoru.ordering_server.domain

import com.byeoru.ordering_server.domain.member.Customer
import javax.persistence.*

@Entity
class Basket(customer: Customer,
             food: Food,
             count: Int) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "basket_id")
     var id: Long? = null
         protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
     var customer: Customer = customer
         protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "food_id")
     var food: Food = food
         protected set

     var count = count
         protected set

    fun updateCount(count: Int) {
        this.count = count
    }

    companion object {
        @JvmStatic
        fun CreateBasket(customer: Customer, food: Food, count: Int): Basket {
            return Basket(customer, food, count)
        }
    }
}